src/components/user/services/user.service.v2.integration.test.js

```diff
@@ -0,0 +1,163 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const { ObjectId } = require('mongoose').Types;
+const appPromise = require('../../../app');
+const testObjects = require('../../../../test/services/helpers/testObjects')(appPromise);
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+describe('user service v2', function test() {
+	let app;
+	let server;
+	let accountModelService;
+	let usersModelService;
+
+	before(async () => {
+		app = await appPromise;
+		server = await app.listen(0);
+		accountModelService = app.service('accountModel');
+		usersModelService = app.service('usersModel');
+	});
+
+	after(async () => {
+		await server.close();
+	});
+
+	const getAdminToken = async (schoolId = undefined) => {
+		const adminUser = await testObjects.createTestUser({ roles: ['administrator'], schoolId });
+		const credentials = { username: adminUser.email, password: `${Date.now()}` };
+		await testObjects.createTestAccount(credentials, 'local', adminUser);
+		const token = await testObjects.generateJWT(credentials);
+		return token;
+	};
+
+	const getPermissionToken = async (schoolId, permissions = []) => {
+		const currentUserRole = `currentUserPermission`;
```

Comment:
this can lead to collisions if the funcion is called multiple times before cleanup, since we then create mutliple roles with the same name. please make sure the name is unique, for example by including a timestamp or id in the string.